src/interaction-handlers/sokoban.ts

```diff
@@ -0,0 +1,83 @@
+import { LanguageKeys } from '#lib/i18n/LanguageKeys';
+import {
+	buildGameControls,
+	buildSokobanGameFromResolvableLevel,
+	buildSokobanGameFromVisualLevel,
+	Direction,
+	encodeResolvableLevel
+} from '#lib/utilities/sokoban';
+import { ActionRowBuilder, ButtonBuilder } from '@discordjs/builders';
+import { InteractionHandler } from '@skyra/http-framework';
+import { getSupportedUserLanguageT } from '@skyra/http-framework-i18n';
+import { ButtonStyle, MessageFlags, type APIButtonComponentWithCustomId } from 'discord-api-types/v10';
+
+const Root = LanguageKeys.Commands.Sokoban;
+
+export class UserHandler extends InteractionHandler {
+	public async run(interaction: InteractionHandler.ButtonInteraction, [directionOrRetry, startTimestampOrEncodedLevel, moves]: Parameters) {
+		const t = getSupportedUserLanguageT(interaction);
+		// if the restart button is pressed, reset the current level
+		if (directionOrRetry === 'retry') {
+			const encodedLevel = startTimestampOrEncodedLevel!.replaceAll('-', '.');
+			const level = buildSokobanGameFromResolvableLevel(encodedLevel).unwrap();
+
+			return interaction.update({
+				content: level.toString(),
+				components: buildGameControls(encodeResolvableLevel(level.components), level.checkPossibleMoves()),
+				flags: MessageFlags.Ephemeral
+			});
+		}
+
+		const startTimestamp = Number(startTimestampOrEncodedLevel);
+
+		// parse components from the emojis in the interaction's message
+		const gameBoardResult = buildSokobanGameFromVisualLevel(interaction.message.content);
+		if (gameBoardResult.isErr()) {
+			return interaction.update({
+				content: t(LanguageKeys.Commands.Sokoban.SokobanInvalidComponent, { value: gameBoardResult.unwrapErr() }),
+				components: [],
+				flags: MessageFlags.Ephemeral
+			});
+		}
+		const level = gameBoardResult.unwrap();
+
+		// execute the user's move
+		const pushedBoxOption = level.executeMove(directionOrRetry);
+		const updatedLevel = level.toString();
+
+		// check win condition, if met, send victory message
+		if (level.checkWinCondition()) {
+			return interaction.update({
+				content: `${t(Root.Victory, { seconds: Math.round((Date.now() - startTimestamp) / 10) / 100, moves })}\n${updatedLevel}`,
```

Comment:
Current code turns `2413.312145` into `2.41`
```suggestion
				content: `${t(Root.Victory, { seconds: Math.round((Date.now() - startTimestamp) * 100) / 100, moves })}\n${updatedLevel}`,
```

Reply:
Actually, it's doing `* 100 / 1000` which makes it `/ 10`, this behaviour is after all intentional.

It's not intuitive at all 😅, maybe we should make this clearer

Reply:
by making more clear, would you like a utility function to do this?

Reply:
Just a comment will suffice